string direction
float32 velocity
float32 limit